cache/lfu.go

```diff
@@ -0,0 +1,112 @@
+// Package lfu (Least Frequently Used) is a type of cache algorithm used to manage memory within a computer.
+// ref: https://en.m.wikipedia.org/wiki/Least_frequently_used
+package cache
+
+import (
+	"fmt"
+
+	"github.com/TheAlgorithms/Go/structure/linkedlist"
+)
+
+// LFUCache is a struct that holds the LFU cache
+type LFUCache struct {
+	lookUp       map[any]*linkedlist.Node[any]
+	frequency    map[any]*linkedlist.Doubly[any]
```

Comment:
You're using `any` here without an actual benefit since you case everything in your implementations. I would suggest using `LFUNode` directly here...

```suggestion
	lookUp       map[any]*linkedlist.Node[LFUNode]
	frequency    map[any]*linkedlist.Doubly[LFUNode]
```